### PR TITLE
macOS install steps and update CI Build

### LINKED ISSUE #58: Need steps for a MacOSX build
Support build steps for macOSX. 

- Update readme with crisp steps for macOSX
- Update GHA to include a mac CI build
- As a first step support a build targetted for macOSX without CUDA

Future requirements (not in this scope) will include:  
- a build targetted for macOSX with CUDA
- a build targetted for Jetson boards but done on MacOSX

### PR DESCRIPTION
**Pull request by dsiddharth2**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #58 

**Description**

Completed - Update readme with crisp steps for macOSX
Completed - Update GHA to include a mac CI build

- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,52 @@ git clone --recursive https://github.com/Apra-Labs/ApraPipes.git
...
+- Update to Xcode 13.2.1 or Recent
+    - Please note you need atleast 40GB of free space on your macPC to update to latest version
+    - Follow this Link to [Install XCode](https://www.freecodecamp.org/news/how-to-download-and-install-xcode/)
+- Install Homebrew
+    ```sh 
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+    ```
+- Install nproc and coreutils
+    ```sh 
+    brew install coreutils
```
Comment (kumaakh) [inline README.md:58]: comment says nproc and coreutils but the command line below shows only coreutils ?

File: README.md [hunk 2]
```diff
@@ -45,6 +45,52 @@ git clone --recursive https://github.com/Apra-Labs/ApraPipes.git
   * Look at the unit_tests/params_test to check for sample usage of parameters in test code
 
 
+## Prerequisites for macOS
+- Update to Xcode 13.2.1 or Recent
+    - Please note you need atleast 40GB of free space on your macPC to update to latest version
+    - Follow this Link to [Install XCode](https://www.freecodecamp.org/news/how-to-download-and-install-xcode/)
+- Install Homebrew
+    ```sh
```
Comment (kumaakh) [inline README.md:53]: what is the use of "sh" here ? I guess it gives that nice copy to clipboard feature ?

File: README.md [hunk 3]
```diff
@@ -45,6 +45,52 @@ git clone --recursive https://github.com/Apra-Labs/ApraPipes.git
...
+    ```
+- Make sure you have path set to CMake,
+    ```sh 
+    PATH="/Applications/CMake.app/Contents/bin":"$PATH"
+    ```
+- Give execute permission to ./build_linux_no_cuda.sh
+    ```sh 
+    chmod +x ./build_linux_no_cuda.sh
+    ```
+- Build Apra Piples No Cuda Version,
```
Comment (kumaakh) [inline README.md:86]: typo in Piples

**Review by kumaakh — CHANGES_REQUESTED** [review]
looks like CI build is failing in configure. was it passing for you on branch ?

File: README.md (hunk 2 above)
Comment (dsiddharth2) [reply README.md:53]: Yes it gives a nice border around the command

**Comment by dsiddharth2** [comment]
> looks like CI build is failing in configure. was it passing for you on branch ?
@kumaakh Yes it was passing for me in my local branch and the mac I was trying to run it on.

**Pull request closed without merging**